makefiles/dependency_resolution.inc.mk

```diff
@@ -2,6 +2,12 @@
 # until no new modules, pkgs, or features are pull in order to catch all
 # transient dependencies
 
+# Locate used modules in $(EXTERNAL_MODULE_DIRS). Internal modules cannot depend on external
+# ones (for obvious reasons) and dependencies for external modules are not supported. Hence,
```

Comment:
> dependencies for external modules are not supported

What do you mean ? I have external modules depending on other external modules and it works fine with your PR.

Reply:
Good you noticed. That is a leftover. I initially wanted to keep them out of the recursive dependency resolution to avoid breakage. But then I checked and noticed that it eas previously supported, so I readded it. I just forgot to update the comment.

Reply:
Fixed